 ~A() {
   if (uncaught_exception ())
     exit (0);
 }